### ISSUE TITLE
Reinitialisiation after structuredClone

### ISSUE THREAD

**Issue by benni336** [iamkun/dayjs#2236]
TLDR:
**Describe the bug**
I want to copy an object storing dayJS objects as properties. In order to get a full copy I use structuredClone()

**Expected behavior**
Retrieve an fully functional Object dayJS after re-initialization.

**Information**
 - Day.js Version Current
 - OS: Browser Firefox Current and NodeJS

**Working Example**
[https://jsfiddle.net/uqm8ts5h/6/](https://jsfiddle.net/uqm8ts5h/6/)

Long Story:
I am a "newbee" to JS and I am fiddeling around with copying an object in order to not do the whole reinitialisation work.
I know I could stringify() and parse(), but then I found the "new" possibility of structuredClone().
So I cloned and reinitialised the object. Sadly dayJS loses it's date. In order to proof it, I made the working example above.
Like in my code all values become NaN after reinitialisation. Please see the Output in the console in the working example

```JS
class anClass {

    constructor(json) {
        this.someDate=dayjs(json.someDate)
        this.someString=json.someString
    }
}

testobject=new anClass({someDate:dayjs('2019-01-25'), somestring: "hello"})
console.log("test1", testobject.someDate)
console.log("test1", testobject.someDate.toString())

testobject2=new anClass( structuredClone(testobject) )

console.log("test2",dayjs.isDayjs(testobject2.someDate))
console.log("test2", testobject2.someDate)
console.log("test2", testobject2.someDate.toString())

```

```
"test1", {
  $D: 25,
  $d: [object Date] { ... },
  $g: function(t,e,n){return O.u(t)?this[e]:this.set(n,t)},
  $H: 0,
  $L: "en",
  $locale: function(){return D[this.$L]},
  $m: 0,
  $M: 0,
  $ms: 0,
  $s: 0,
[...]
}
"test1", "Thu, 24 Jan 2019 23:00:00 GMT"

"test2", true // <- dayjs.isDayjs()
"test2", {
  $D: NaN,
  $d: [object Date] { ... },
  $g: function(t,e,n){return O.u(t)?this[e]:this.set(n,t)},
  $H: NaN,
  $L: "en",
  $locale: function(){return D[this.$L]},
  $m: NaN,
  $M: NaN,
  $ms: NaN,
  $s: NaN,
[...]

```

Checking `JS console.log( structuredClone(testobject) )` seems to return a stripped dayJS-Object. Maybe there is a possibility to tell dayJS to return some Time-String that can than be reinitialised with dayJS() ?

Hope it is just my lack of knowledge.
Best

**Comment by tylersouthard** [comment]
I'm curious about this as well.

**Comment by FoxxMD** [comment]
Also running into this issue when trying to use [in memory caching](https://cacheable.org/docs/memory/) that clones value on set, using structuredClone.
